iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.cs

```diff
@@ -1,101 +1,108 @@
-                  {
-                    Console.WriteLine("  {0}: {1}", prop.Key, prop.Value);
-                  }
-                }
-            }
-        }
-
-        private static async Task Main(string[] args)
-        {
-            Console.WriteLine("IoT Hub Quickstarts - Read device to cloud messages. Ctrl-C to exit.\n");
-
-            // Create an EventHubClient instance to connect to the
-            // IoT Hub Event Hubs-compatible endpoint.
-            var connectionString = new EventHubsConnectionStringBuilder(new Uri(s_eventHubsCompatibleEndpoint), s_eventHubsCompatiblePath, s_iotHubSasKeyName, s_iotHubSasKey);
-            s_eventHubClient = EventHubClient.CreateFromConnectionString(connectionString.ToString());
-
-            // Create a PartitionReciever for each partition on the hub.
-            var runtimeInfo = await s_eventHubClient.GetRuntimeInformationAsync();
-            var d2cPartitions = runtimeInfo.PartitionIds;
-
-            CancellationTokenSource cts = new CancellationTokenSource();
-
-            Console.CancelKeyPress += (s, e) =>
-            {
-                e.Cancel = true;
-                cts.Cancel();
-                Console.WriteLine("Exiting...");
-            };
-
-            var tasks = new List<Task>();
-            foreach (string partition in d2cPartitions)
-            {
-                tasks.Add(ReceiveMessagesFromDeviceAsync(partition, cts.Token));
-            }
-
-            // Wait for all the PartitionReceivers to finsih.
-            Task.WaitAll(tasks.ToArray());
-        }
-    }
-}
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// This application uses the Azure Event Hubs Client for .NET
+// For samples see: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/README.md
+// For documentation see: https://docs.microsoft.com/azure/event-hubs/
+
+using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+
+namespace read_d2c_messages
+{
+    class ReadDeviceToCloudMessages
+    {
+        // Event Hub-compatible endpoint
+        // az iot hub show --query properties.eventHubEndpoints.events.endpoint --name {your IoT Hub name}
+        private const string EventHubsCompatibleEndpoint = "{your Event Hubs compatible endpoint}";
+
+        // Event Hub-compatible name
+        // az iot hub show --query properties.eventHubEndpoints.events.path --name {your IoT Hub name}
+        private const string EventHubName = "{your Event Hubs compatible name}";
+
+        // az iot hub policy show --name service --query primaryKey --hub-name {your IoT Hub name}
+        private const string IotHubSasKey = "{your service primary key}";
+        private const string IotHubSasKeyName = "{your service primary key's name}";

```

Comment:
@jebrando Can you please just set this constant to "service". That's the policy that we're getting in the CLI command.

Reply:
Not sure if you intended to reference me... but, if so, I'm not sure that I follow the ask.  Would you be so kind as to clarify?

Reply:
Yup please set the value to "service". 

IoT Hub exposes a bunch of different shared access policies that grant different privileges, i.e. iothubowner, service, device, registryRead, registryReadWrite. We want people to use the least privileged policy for whatever ops they are wanting to perform. In this case, that would be the "service" policy. I'm honestly not really sure how/whether that has any effect on the privileges granted by EH. I think I started a thread at one point, but dropped it.

Reply:
This has been updated.

Reply:
Great. Thanks!